src/plugins/vis_type_xy/public/plugin.ts

```diff
@@ -77,8 +77,9 @@ export class VisTypeXyPlugin
     if (!core.uiSettings.get(LEGACY_CHARTS_LIBRARY, false)) {
       setUISettings(core.uiSettings);
       setThemeService(charts.theme);
-      setColorsService(charts.legacyColors);
-
+      charts.palettes.getPalettes().then((palettes) => {
```

Comment:
This function should not be called directly in a plugin lifecycle because it's meant to enable lazy loading (if it's loaded every time the plugin is loaded, that's pretty much defeated).

It should be called if it's certain a chart will be rendered very soon (e.g. in the renderer or behind some existing lazy loading barrier)